cli/cliui/cliui.go

```diff
@@ -117,48 +101,150 @@ func ifTerm(fmt pretty.Formatter) pretty.Formatter {
 }
 
 func init() {
+	color = termenv.NewOutput(os.Stdout).ColorProfile()
+	if flag.Lookup("test.v") != nil {
+		// Use a consistent colorless profile in tests so that results
+		// are deterministic.
+		color = termenv.Ascii
+	}
+
 	// We do not adapt the color based on whether the terminal is light or dark.
 	// Doing so would require a round-trip between the program and the terminal
 	// due to the OSC query and response.
-	DefaultStyles = Styles{
-		Code: pretty.Style{
-			ifTerm(pretty.XPad(1, 1)),
-			pretty.FgColor(Red),
-			pretty.BgColor(color.Color("#2c2c2c")),
-		},
-		DateTimeStamp: pretty.Style{
-			pretty.FgColor(color.Color("#7571F9")),
-		},
-		Error: pretty.Style{
-			pretty.FgColor(Red),
-		},
-		Field: pretty.Style{
-			pretty.XPad(1, 1),
-			pretty.FgColor(color.Color("#FFFFFF")),
-			pretty.BgColor(color.Color("#2b2a2a")),
-		},
-		Keyword: pretty.Style{
-			pretty.FgColor(Green),
-		},
-		Placeholder: pretty.Style{
-			pretty.FgColor(color.Color("#4d46b3")),
-		},
-		Prompt: pretty.Style{
-			pretty.FgColor(color.Color("#5C5C5C")),
-			pretty.Wrap("> ", ""),
-		},
-		Warn: pretty.Style{
-			pretty.FgColor(Yellow),
+	DefaultStyles = Styles{Wrap: pretty.Style{pretty.LineWrap(80)}}
+
+	defaultTheme := userThemeConfig{
+		Colors: map[string]string{
+			"green":  "#04B575",
+			"red":    "#ED567A",
+			"yellow": "#ECFD65",
+			"blue":   "#5000ff",
 		},
-		Wrap: pretty.Style{
-			pretty.LineWrap(80),
+		Styles: &userThemeStyles{
+			Code: &userThemeStyle{
+				FgColor: "$red",
+				BgColor: "#2C2C2C",
+			},
+			DateTimeStamp: &userThemeStyle{
+				FgColor: "#7571F9",
+			},
+			Error: &userThemeStyle{
+				FgColor: "$red",
+			},
+			Field: &userThemeStyle{
+				FgColor: "#FFFFFF",
+				BgColor: "#2B2A2A",
+			},
+			FocusedPrompt: &userThemeStyle{
+				FgColor: "$blue",
+			},
+			Keyword: &userThemeStyle{
+				FgColor: "$green",
+			},
+			Placeholder: &userThemeStyle{
+				FgColor: "#4d46b3",
+			},
+			Prompt: &userThemeStyle{
+				FgColor: "#5C5C5C",
+			},
+			Warn: &userThemeStyle{
+				FgColor: "$yellow",
+			},
 		},
 	}
 
-	DefaultStyles.FocusedPrompt = append(
-		DefaultStyles.Prompt,
-		pretty.FgColor(Blue),
-	)
+	configDir := config.DefaultDir()
+	if dir := os.Getenv("CODER_CONFIG_DIR"); dir != "" {
+		configDir = dir
+	}
+
+	if theme, err := os.ReadFile(path.Join(configDir, "theme.toml")); err == nil {
+		_ = toml.Unmarshal(theme, &defaultTheme)
```

Comment:
Sure thing! My thinking was that an invalid theme file shouldn't break the CLI in any way, but emitting a warning log wouldn't cause any harm.

Reply:
I agree, for better future compatibility it's probably best to just warn. Having to fix an error just as you're trying to get something done is _not nice_.

Reply:
Instead of swallowing the error we provide a nice message (provided by the go-toml library used).

<img width="955" alt="Screenshot 2024-09-12 at 15 49 03" src="https://github.com/user-attachments/assets/f9eb42db-5ab8-4888-83d0-b64233b82df8">